src/containers/TransactionPage/TransactionPage.duck.js

```diff
@@ -427,24 +428,40 @@ export const fetchTransaction = (id, txRole, config) => (dispatch, getState, sdk
         console.log(`transaction process (${processName}) was not recognized`);
       }
 
-      const canFetchListing = listing && listing.attributes && !listing.attributes.deleted;
-      if (canFetchListing) {
-        return sdk.listings.show({
-          id: listingId,
-          include: ['author', 'author.profileImage', 'images'],
-          ...getImageVariants(config.layout.listingImage),
-        });
-      } else {
-        return response;
-      }
+      // API does not allow fetching transaction.listing.author, so we will
+      // set the relationship manually based on the transaction provider's id.
+      // The full resource is already available in the response.data.included array,
+      // so we only need to add the relationship reference in the included listing resource.
+      const { included } = response.data;
+      const includedListingIdx = included.findIndex(inc => inc.type === 'listing');
+
+      included[includedListingIdx] = {
+        ...included[includedListingIdx],
+        relationships: {
+          ...included[includedListingIdx].relationships,
+          author: {
+            data: {
+              id: provider.id,
+              type: 'user',
+            },
+          },
+        },
+      };
```

Comment:
Turn this into a helper function like listingRelationship, which is defined on line 377.

Reply:
Extracted to a separate helper.